### ISSUE TITLE
Standardised Soft Toggle

### ISSUE THREAD

**Issue by Aktimoose** [geode-sdk/geode#655]
### Geode Version

v2.0.0beta23

### Related Problem

Everytime I need to toggle a mod on or off, it is frustrating that I need to restart the whole game for it most of the time.

### Your Solution

A few mods have an option within them to toggle the entire functionality on and off for the mod without restarting, such as Death Markers by Adam_729. It would be good if this was encouraged for most mods to adopt by having the functionality of the checkmark do a soft toggle instead of a hard toggle (with hard toggling still being accessible from the View menu).

### Additional Information

This could be done by having the checkmark set a flag that the mod can access to see if it is toggled on/off. There should be an option for the mod to opt into this behavior by putting something into the mod.json, and if the mod did not opt in, do the hard toggle as usual.

**Comment by cursefroge** [comment]
Good idea! Maybe events for saving/loading, rather than setting a flag? Kind of like [Sponge's RefreshGameEvent](https://docs.spongepowered.org/stable/en/plugin/event/listeners.html#game-reload)?
